components/post_view/acknowledgements/post_acknowledgements.tsx

```diff
@@ -0,0 +1,147 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React, {memo, useState} from 'react';
+import {FormattedMessage} from 'react-intl';
+import classNames from 'classnames';
+import {useDispatch} from 'react-redux';
+import {
+    useFloating,
+    autoUpdate,
+    offset,
+    flip,
+    shift,
+    useHover,
+    useRole,
+    useInteractions,
+    safePolygon,
+    FloatingFocusManager,
+    useId,
+} from '@floating-ui/react-dom-interactions';
+
+import {CheckCircleOutlineIcon} from '@mattermost/compass-icons/components';
+
+import {acknowledgePost, unacknowledgePost} from 'mattermost-redux/actions/posts';
+
+import {Post, PostAcknowledgement} from '@mattermost/types/posts';
+import {UserProfile} from '@mattermost/types/users';
+
+import PostAcknowledgementsUserPopover from './post_acknowledgements_users_popover';
+
+import './post_acknowledgements.scss';
+
+type Props = {
+    acknowledgedAt: number;
+    currentUserId: UserProfile['id'];
+    hasReactions: boolean;
+    list?: Array<{user: UserProfile; acknowledgedAt: PostAcknowledgement['acknowledged_at']}>;
+    postId: Post['id'];
+}
+
+function moreThan5minAgo(time: number) {
+    const now = new Date().getTime();
+    return now - time > 5 * 60 * 1000;
+}
+
+function PostAcknowledgements({
+    acknowledgedAt,
+    currentUserId,
+    hasReactions,
+    list,
+    postId,
+}: Props) {
+    const dispatch = useDispatch();
+    const [open, setOpen] = useState(false);
+
+    const {x, y, reference, floating, strategy, context} = useFloating({
+        open,
+        onOpenChange: setOpen,
+        placement: 'top-start',
+        whileElementsMounted: autoUpdate,
+        middleware: [
+            offset(5),
+            flip({fallbackPlacements: ['bottom', 'right'], padding: 5}),
+            shift(),
+        ],
+    });
+
+    const headingId = useId();
+
+    const {getReferenceProps, getFloatingProps} = useInteractions([
+        useHover(context, {
+            enabled: list && list.length > 0,
+            delay: {
+                open: 300,
+                close: 0,
+            },
+            handleClose: safePolygon(),
+        }),
+        useRole(context),
+    ]);
+
+    const handleClick = () => {
+        if (acknowledgedAt) {
+            dispatch(unacknowledgePost(postId));
+        } else {
+            dispatch(acknowledgePost(postId));
+        }
+    };
+
+    const button = (
+        <button
+            ref={reference}
+            onClick={handleClick}
+            disabled={Boolean(acknowledgedAt) && moreThan5minAgo(acknowledgedAt)}
```

Comment:
I don't think this needs to be changed, but I just wanted to note here that this won't update on its own once 5 minutes pass. Something else will have to cause React to update first. That's probably fine though since we have enough going on that something will trigger an update cycle shortly anyway though, and I doubt people will be specifically timing this or anything.

Reply:
yep that's true, but even then if the user clicks the api will reject, and the UI will be updated (I think, if not we can trigger an update then) so the button will be disabled afterwards